server/app/services/applications/ApplicationNotFoundException.java

```diff
@@ -0,0 +1,7 @@
+package services.applications;
+
+public final class ApplicationNotFoundException extends Exception {
```

Comment:
```suggestion
public final class ApplicationNotFoundException extends RuntimeException {
```
Don't extend directly from Exception. That create checked exceptions requiring you to either have to have excessive try/catch or add `throws blah` on any method that doesn't handle it.